### PR TITLE
feat: support all possible elasticapm.* settings

### PR DESCRIPTION
**Pull request by karfau**
The main purpose is to support the `elasticapm.collect_local_variables` setting, but instead of just enabling this setting I found a way to enable all possible settings.

In order to have a reliable setup for changes I also applied the following changes:

- updated pre-commit regarding broken URLs
- added bettermarks/.github git-hooks
- run black on all code
- update pyramid to 1.10 and mention the reason in the readme
- add contributing section to readme
- update install instructions
- update repo URL in setup.py
- drop scheduled trigger and update actions

https://bettermarks.atlassian.net/browse/BM-61809

### CODE REVIEW

**Review by saurabh2590 — DISMISSED** [review]
LGTM. 
Nice to have: We can consider creating pull request to open project but we don't have to.

**Review by saurabh2590 — APPROVED** [review]

File: src/pyramid_elasticapm/__init__.py [hunk 1]
```diff
@@ -26,23 +26,21 @@ def elasticapm_instrument(event):
...
-            'ENVIRONMENT': settings['elasticapm.environment'],
-            'TRANSACTION_SAMPLE_RATE': settings['elasticapm.transaction_sample_rate'],
+            key.replace(ELASTICAPM_PREFIX, '').upper(): value
+            for key, value in settings.items()
+            if key.startswith(ELASTICAPM_PREFIX)
         }
-        if settings.get('elasticapm.transactions_ignore_patterns', ''):
-            config['TRANSACTIONS_IGNORE_PATTERNS'] = settings[
-                'elasticapm.transactions_ignore_patterns'
-            ].split()
```
Comment (cakemanny) [inline src/pyramid_elasticapm/__init__.py:45]: I don't understand this...

This looks like the setting needs to be split into a list. I can't see how this behaviour is preserved....
Could you explain it to me? _maybe it's a Monday morning thing... :)_

Comment (karfau) [reply src/pyramid_elasticapm/__init__.py:45]: You are right. Will fix it.

Comment (karfau) [reply src/pyramid_elasticapm/__init__.py:45]: I nearly had the implementation ready, just to find that the elastic client already handles that case of an unsplit value:
image
so I will not add any additional logic to this PR

**Review by cakemanny — APPROVED** [review]
Nice work 🙌

**Comment by karfau** [comment]
@saurabh2590 @cakemanny since we do not pin the version in backend it will be available with the next backend change, right?

**Comment by cakemanny** [comment]
> @saurabh2590 @cakemanny since we do not pin the version in backend it will be available with the next backend change, right?

I think you're correct there

**Pull request merged**